app/code/Magento/Catalog/Model/ProductRepository.php

```diff
@@ -23,6 +23,7 @@
 use Magento\Framework\Exception\NoSuchEntityException;
 use Magento\Framework\Exception\StateException;
 use Magento\Framework\Exception\ValidatorException;
+use Magento\ConfigurableProduct\Model\Product\Type\Configurable;
```

Comment:
You cannot refer to `ConfigurableProduct` module in `Catalog`.

Reply:
As far as I remember I already asked in previous PR but it was force pushed or something.

I don't see a good way to move such addition to `ConfigurableProduct` module and I don't see disadvantages of having `min_price` for simple product (for example, if simple product has non-free custom options, price without any of them would be minimal).

@vrann thoughts?